src/Microsoft.DocAsCode.MarkdigEngine.Extensions/MarkdownContext.cs

```diff
@@ -10,6 +10,11 @@ namespace Microsoft.DocAsCode.MarkdigEngine.Extensions
 
     public class MarkdownContext
     {
+        /// <summary>
+        /// Log delegate
+        /// </summary>
+        public delegate void LogActionDelegate(string message, string phase = null, string file = null, string line = null, string code = null);
```

Comment:
To align with v3 logging, this could be `(string code, string message, string file = null, int line)`. `code` is mandatory.